dsgrid/helpers.py

```
@@ -73,10 +73,12 @@ def is_s3(self):
         return str(self.filepath).startswith("s3://")
 
     def __enter__(self):
-        return self._f.__enter__()
+        return self._f
 
     def __exit__(self, exc, value, tb):
-        result = self._f.__exit__(exc, value, tb)
+        self._f.close()
         if self.is_s3:
-            self._s3p.__exit__(exc, value, tb)
-        return result
\ No newline at end of file
+            self._s3p.close()
+        if exc is not None:
+            raise
```

Do you need to raise the exception here? I'm asking because I haven't done that, and maybe I'm doing it incorrectly.

---

I'm not sure--I pulled it from @MRossol's example. Michael, willing to enlighten us?

---

Good question, I was under the impression that the close captures the exc, value, and traceback. Its unclear if it would raise the exc automatically hence checking for one and raising it explicitly.

---

I did a quick test on one of my context managers and the exception was still thrown. I think it is provided in case you want to do something with it.

---

Good to know!